micropip/_micropip.py

```diff
@@ -692,3 +695,72 @@ def _list():
             source_ = pkg_source
         packages[name] = PackageMetadata(name=name, version=version, source=source_)
     return packages
+
+
+def add_mock_package(name, version, *, modules=None):
+    """
+       Add a mock version of a package to the package dictionary.
+
+       This means that if it is a dependency, it is skipped on install.
+
+       By default a single empty module is installed with the same
+       name as the package. You can alternatively give one or more modules to make a
+       set of named modules.
+
+       The modules parameter is a dictionary mapping module name to module text.
+
+        e.g.
+       ``
+       {
+          mylovely_module:'''
+            def module_method(an_argument):
+                print("This becomes a module level argument")
+            module_value = "this value becomes a module level variable"
+
+            print("This is run on import of module")
+          '''
+       }
+
+
+       ``
+
+    Parameters
+    ----------
+    name : ``str ``
+
+        Package name to add
+
+    modules : ``Optional(dict) ``
+
+        Dictionary of module_name:string pairs.
+        The string contains the source of the mock module or is blank for
+        an empty module.
+    """
+    if modules is None:
+        # make a single mock module with this name
+        modules = {name: ""}
+    # make empty mock modules with the requested names in user site packages
+    site_packages = Path(site.getusersitepackages())
+    metadata_file = site_packages / (name + "-" + version + ".egg-info")
+    with open(metadata_file, "w") as mf:
+
+        mf.write(
+            f"""Metadata-Version: 1.1
+Name: {name}
+Version: {version}
+Summary: {name} mock module generated by micropip
+Author-email: {name}@micro.pip.non-working-fake-host
+"""
+        )
+        for n, s in modules.items():
+            mf.write(f"Provides:{n}\n")
+
+            if s is None:
+                s = ""
+            s = dedent(s)
+            path_parts = n.split(".")
+            dir_path = Path(site_packages, *path_parts)
+            os.makedirs(dir_path, exist_ok=True)
```

Comment:
Maybe we should raise error if a directory exists, because it would mean that the package is already installed somehow.